sdk/core/azure-core/inc/azure/core/internal/diagnostics/log.hpp

```diff
@@ -11,24 +11,24 @@
 
 namespace Azure { namespace Core { namespace Diagnostics { namespace _internal {
   class Log final {
-    using LogLevelInt = std::underlying_type<Logger::Level>::type;
-
     static_assert(
-        std::is_same<int, LogLevelInt>::value == true && ATOMIC_INT_LOCK_FREE == 2,
+        std::is_same<int, std::underlying_type<Logger::Level>::type>::value == true,
         "Logger::Level values must be representable as lock-free");
 
+    static_assert(ATOMIC_INT_LOCK_FREE == 2, "atomic<int> must be lock-free");
+
     static_assert(ATOMIC_BOOL_LOCK_FREE == 2, "atomic<bool> must be lock-free");
 
     static AZ_CORE_DLLEXPORT std::atomic<bool> g_isLoggingEnabled;
-    static AZ_CORE_DLLEXPORT std::atomic<LogLevelInt> g_logLevel;
+    static AZ_CORE_DLLEXPORT std::atomic<Logger::Level> g_logLevel;
```

Comment:
Still, no explicit guarantee that if `atomic<int>` is lock-free, that `atomic<Logger::Level>` is also lock-free, even though the underlying type is `int`.